geomstats/geometry/product_manifold.py

```diff
@@ -208,8 +210,11 @@ def project_from_product(self, point):
                 for j in range(len(self.factors))
             ]
 
-        else:
+        elif len(self.shape) <= 2:
             projected_points = [point[..., j, :] for j in range(len(self.factors))]
+        else:
+            point = gs.reshape(point, (-1,) + self.shape)
+            projected_points = [point[:, j, ...] for j in range(len(self.factors))]
```

Comment:
no need for reshape: `point[..., j, :, :]` should work.

Reply:
Hello @lpereira95,
Yes, using `point[..., j, :, :]` is working when `len(self.shape) == 2`.
My solution (reshaping the data) should be working when `len(self.shape) >= 2`.
Which solution would you like me to keep?